src/devinterp/slt/norms.py

```diff
@@ -26,4 +27,28 @@ def _flatten_weights(self, model: nn.Module):
         params = [torch.flatten(param) for param in params]
         params = torch.cat(params)
         return params
-    
\ No newline at end of file
+
+
+class GradientNorm(ChainCallback):
+    def __init__(self, num_chains, num_draws, device='cpu'):
+        self.gradient_norms = torch.tensor((num_chains, num_draws), dtype=torch.float32).to(device)
+        self.device = device
+    
+    def __call__(self, chain: int, draw: int, model: nn.Module):
+        self.update(chain, draw, model)
+    
+    def update(self, chain: int, draw: int, model: nn.Module):
+        grads = self._flatten_grads(model)
+        self.gradient_norms[chain, draw] = torch.linalg.vector_norm(grads, ord=2)
+
+    def sample(self):
+        return {
+            "gradient_norm/trace": self.gradient_norms.cpu().numpy()
+        }
+    
+    def _flatten_grads(self, model: nn.Module):
```

Comment:
See my comment on #49. It'd be more efficient to avoid having to create a new tensor here